src/main/api-utilities/ScenarioAttributeVerifier.js

```diff
@@ -0,0 +1,67 @@
+import {
+    AttributesVerifier,
+    ValueVerifier,
+    ActorVerifier,
+    ActionVerifier
+} from './../../main/api-utilities/PropertyVerifier';
+
+export class ScenarioAttributeVerifier {
+    constructor(
+        attributeVerifier = AttributesVerifier,
+        valueVerifier = ValueVerifier,
+        actorVerifier = ActorVerifier,
+        actionVerifier = ActionVerifier
+    ) {
+        this.attributeVerifier = attributeVerifier;
+        this.valueVerifier = valueVerifier;
+        this.actorVerifier = actorVerifier;
+        this.actionVerifier = actionVerifier;
+    }
+
+    check(scenario) {
+        return this.attributeVerifier.check(scenario, ['actors', 'actions', 'scenes']);
+    }
+}
+
+export class ActorAttributeVerifier extends ScenarioAttributeVerifier {
+
+    check(actor) {
+        var result = this.attributeVerifier.check(actor, ['health', 'name', 'weapon', 'type', 'position']);
+        if (result == true) result = this.valueVerifier.check(actor.position.xPos, 0, 100);
+        if (result == true) result = this.valueVerifier.check(actor.position.yPos, 0, 100);
+        return result;
+    }
+}
+
+export class ActionAttributeVerifier extends ScenarioAttributeVerifier {
+
+    check(scenario, action) {
+        var result = this.attributeVerifier.check(action, ['actor', 'action', 'scenes']);
+        if (result == true) result = this.actionVerifier.check(action);
+        if (result == true) result = this.actorVerifier.check(scenario, action.actor);
+        if (result == true) result = this.valueVerifier.check(action.scenes, 0, 9999);
+
```

Comment:
Here is an unused space, keep in mind.

But what a nice implementation for the verification process, very good job